hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/FileSinkBuilder.java

```diff
@@ -120,17 +125,41 @@
      *      guarantee. If false, sink's guarantee will be at-least-once
      *      even if job's is exactly-once
      * @return this instance for fluent API
+     *
+     * @since 4.0
      */
     public FileSinkBuilder<T> exactlyOnce(boolean enable) {
         exactlyOnce = enable;
         return this;
     }
 
+    /**
+     * Sets if files are in a shared storage visible to all members. Default
+     * value is {@code false}.
+     * <p>
+     * If {@code sharedFileSystem} is {@code true}, Jet will assume all members
+     * see the same files. Otherwise it will assume that only processors on the
+     * local member them.
+     * <p>
+     * If you start all the members on a single machine (such as for
+     * development), set this property to true. If you have multiple machines
+     * with multiple members each and the directory is not a shared storage,
+     * it's not possible to configure the file reader correctly - use only one
+     * member per machine.
+     *
+     * @since 4.0
+     */
+    @Nonnull
+    public FileSinkBuilder<T> sharedFileSystem(boolean sharedFileSystem) {
```

Comment:
can't we always assume it's shared, why require the user to specify it or not? It makes sense for source, but not for sink. Even if it's not shared, you want distinct files you can merge later.